CONTRIBUTING.md

```diff
@@ -39,6 +41,27 @@ maximize the chances of your PR being merged.
 * All deprecations/breaking changes will be clearly listed in [DEPRECATED.md](DEPRECATED.md).
 * All deprecations/breaking changes will be announced to the
   [envoy-announce](https://groups.google.com/forum/#!forum/envoy-announce) email list.
+* Protobuf configuration in an alpha namespace, e.g. `v2alpha`, do not have any
+  restrictions on breaking changes. They may be freely modified, together with
+  their respective features.
+* Configuration in the `v2` namespace are considered stable and subject to the
+  above policy. They are
+  [frozen](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/v2_overview#status).
+  There will be no changes leading to wire incompatibility (as describe in the
+  [API style guide](api/STYLE.md)), however fields may be deprecated over time.
+  When a field is deprecated, it will follow the deprecation release cycle
+  described above.
+* No configuration field or message in the `v2` namespace may be deprecated
+  unless there is a corresponding semantic equivalent available to replace it.
+  The litmus test is to imagine that a stateless translation tool exists that
+  could convert from the earlier API to the new API. A field may be deprecated
+  if this tool would be able to perform the conversion. For example, removing a
+  field to describe HTTP/2 window settings is valid if a more comprehensive
+  HTTP/2 protocol options field is being introduced to replace it.
```

Comment:
Does this cover the case where we outright delete something without replacement? Is it perhaps worth it to specify this as a specific thing this bullet covers?

Reply:
Yep, clarified.